src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -736,34 +711,51 @@ private void WriteGreenFactory(Node nd, bool withSyntaxFactoryContext = false)
 
                 if (!IsAnyList(field.Type) && !IsOptional(field))
                 {
-                    WriteLine("      if ({0} == null)", CamelCase(field.Name));
-                    WriteLine("        throw new ArgumentNullException(nameof({0}));", CamelCase(field.Name));
+                    WriteLine("if ({0} == null) throw new ArgumentNullException(nameof({0}));", CamelCase(field.Name));
```

Comment:
we were inconsistent between single-line if checks, or an if with an indented next statement.  I tried to consistently use the same pattern everywhere.

Reply:
Shouldn't the code produce be `if ({0} is null) throw` ?

Reply:
There is no operator overloading with these types.

Reply:
Doesn't @jaredpar say `... is null` the definitive null check?

Reply:
i don't see this discussion as being valuable for the core issues ths PR is attempting to solve.  If you really want to make this change feel free to submit your own PR here.